pkg/controller/storagecluster/reconcile.go

```diff
@@ -275,6 +290,55 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 	return reconcile.Result{}, phaseErr
 }
 
+// getNodeTopologyMap
+func (r *ReconcileStorageCluster) getNodeTopologyMap(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	nodes := &corev1.NodeList{}
+	nodeListOptions := client.ListOptions{}
+	nodeListOptions.SetLabelSelector(nodeAffinityKey)
+
+	err := r.client.List(context.TODO(), &nodeListOptions, nodes)
+	if err != nil {
+		return err
+	}
+
+	topologyMap := ocsv1.NodeTopologyMap{}
+
+	for _, node := range nodes.Items {
+		labels := node.Labels
+		for _, label := range labels {
+			for _, key := range validTopologyLabelKeys {
+				if strings.Contains(label, key) {
+					value := labels[label]
+					if !topologyMap.Contains(label, value) {
+						topologyMap.Add(label, value)
+					}
+				}
+			}
+		}
+
+	}
+
+	sc.Status.NodeTopologies = topologyMap
```

Comment:
This will just update the object and not write it to the server. (if that is intended)
Use `Client().Status().Update()` instead to update object definition on server.

Reply:
the storagecluster is updated once at the end of the reconcile loop